base_tier_validation/models/tier_validation.py

```diff
@@ -0,0 +1,870 @@
+                    ("group_ids", "=", False),
+                    *(extra_domain or []),
+                ]
+            )
+            .mapped("field_ids.name")
+        )
+        if add_base_exceptions:
+            exception_fields += BASE_EXCEPTION_FIELDS
+        return list(set(exception_fields))
+
+    @api.model
+    def _get_all_validation_exceptions(self):
+        """Extend for more field exceptions to be written when on the entire
+        validation process."""
+        return self._get_validation_exceptions()
+
+    @api.model
+    def _get_under_validation_exceptions(self):
+        """Extend for more field exceptions to be written under validation."""
+        return self._get_validation_exceptions(
+            extra_domain=[("allowed_to_write_under_validation", "=", True)]
+        )
+
+    @api.model
+    def _get_after_validation_exceptions(self):
+        """Extend for more field exceptions to be written after validation."""
+        return self._get_validation_exceptions(
+            extra_domain=[("allowed_to_write_after_validation", "=", True)]
+        )
+
+    def _check_allow_write_under_validation(self, vals):
+        """Allow to add exceptions for fields that are allowed to be written
+        even when the record is under validation."""
+        exceptions = self._get_under_validation_exceptions()
+        for val in vals:
+            if val not in exceptions:
+                return False
+        return True
+
+    def _check_allow_write_after_validation(self, vals):
+        """Allow to add exceptions for fields that are allowed to be written
+        even when the record is after validation."""
+        exceptions = self._get_after_validation_exceptions()
+        for val in vals:
+            if val not in exceptions:
+                return False
+        return True
+
+    def _get_fields_to_write_validation(self, vals, records_exception_function):
+        """Not allowed fields to write when validation depending
+        on the given function."""
+        exceptions = records_exception_function()
+        not_allowed_fields = []
+        for val in vals:
+            if val not in exceptions:
+                not_allowed_fields.append(val)
+        if not not_allowed_fields:
+            return []
+
+        not_allowed_field_names, allowed_field_names = [], []
+        for fld_name, fld_data in self.fields_get(
+            not_allowed_fields + exceptions
+        ).items():
+            if fld_name in not_allowed_fields:
+                not_allowed_field_names.append(fld_data["string"])
+            else:
+                allowed_field_names.append(fld_data["string"])
+        return allowed_field_names, not_allowed_field_names
+
+    def write(self, vals):
+        self._tier_validation_check_state_on_write(vals)
+        self._tier_validation_check_write_allowed(vals)
+        self._tier_validation_check_write_remove_reviews(vals)
+        return super().write(vals)
+
+    def _write(self, vals):
```

Comment:
```suggestion
    def _write(self, vals):
```
i see you have updated the addition of overriding the `_write_multi` function which was introduced in version 18.0, but strangely the `_write` function will execute [_write_multi](https://github.com/odoo/odoo/blob/066a63c0bbedd305e13e6c7bbee2089ba1167acf/odoo/models.py#L4786) then why should we keep the overriding logic in this function